v0/destinations/snapchat_conversion/transform.js

```diff
@@ -0,0 +1,385 @@
+
+    // Batch event into dest batch structure
+    chunk.forEach(ev => {
+      batchResponseList.push(ev.message.body.JSON);
+      metadata.push(ev.metadata);
+    });
+
+    batchEventResponse.batchedRequest.body.JSON_ARRAY = {
+      batch: JSON.stringify(batchResponseList)
+    };
+
+    batchEventResponse.batchedRequest.endpoint = ENDPOINT;
+    batchEventResponse.batchedRequest.headers = {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${apiKey}`
+    };
+    batchEventResponse = {
+      ...batchEventResponse,
+      metadata,
+      destination
+    };
+    batchedResponseList.push(
+      getSuccessRespEvents(
+        batchEventResponse.batchedRequest,
+        batchEventResponse.metadata,
+        batchEventResponse.destination,
+        true
+      )
+    );
+  });
+
+  return batchedResponseList;
+}
+
+function getEventChunks(event, eventsChunk) {
+  // build eventsChunk of MAX_BATCH_SIZE
+  eventsChunk.push(event);
+}
+
+const processRouterDest = async inputs => {
+  if (!Array.isArray(inputs) || inputs.length <= 0) {
+    const respEvents = getErrorRespEvents(null, 400, "Invalid event array");
+    return [respEvents];
+  }
+
+  const trackResponseList = []; // list containing single track event in batched format
+  let eventsChunk = []; // temporary variable to divide payload into chunks
+  const arrayChunks = []; // transformed payload of (2000) batch size
+  const errorRespList = [];
+  await Promise.all(
+    inputs.map(async (event, index) => {
+      try {
+        if (event.message.statusCode) {
+          // already transformed event
+          getEventChunks(event, eventsChunk);
+          // slice according to batch size
+          if (
+            eventsChunk.length &&
+            (eventsChunk.length >= MAX_BATCH_SIZE ||
+              index === inputs.length - 1)
+          ) {
+            arrayChunks.push(eventsChunk);
+            eventsChunk = [];
+          }
+        } else {
+          // if not transformed
+          getEventChunks(
+            {
+              message: await process(event),
+              metadata: event.metadata,
+              destination: event.destination
+            },
+            trackResponseList,
+            eventsChunk
+          );
+          // slice according to batch size
+          if (
+            eventsChunk.length &&
+            (eventsChunk.length >= MAX_BATCH_SIZE ||
+              index === inputs.length - 1)
+          ) {
+            arrayChunks.push(eventsChunk);
+            eventsChunk = [];
+          }
+        }
+      } catch (error) {
+        errorRespList.push(
+          getErrorRespEvents(
+            [event.metadata],
+            error.response ? error.response.status : 400,
+            error.message || "Error occurred while processing payload."
+          )
+        );
+      }
+    })
+  );
+
+  let batchedResponseList = [];
+  if (arrayChunks.length) {
+    batchedResponseList = await batchEvents(arrayChunks);
```

Comment:
we are waiting this function but `batchEvents` is not defined as async, check carefully and make sure no errors are happening regarding that

Reply:
have you checked this? if the function is not sync, is there any specific reason that you want to make it await? cc @ItsSudip @utsabc

Reply:
We can remove the await for this.